ocaml/xapi/xapi_vusb.ml

```diff
@@ -32,6 +32,12 @@ let create ~__context ~vM ~uSB_group ~other_config =
       another vusb can not create with the same USB_group. *)
   if vusbs <> [] then
     raise (Api_errors.Server_error(Api_errors.too_many_vusbs, ["1"]));
+  (* We won't attach VUSB when VM ha_restart_priority is set to 'restart'  *)
+  let ha_restart_priority = Db.VM.get_ha_restart_priority ~__context ~self:vM in
+  match ha_restart_priority with
+    | Some Constants.ha_restart -> raise (Api_errors.Server_error(Api_errors.operation_not_allowed,
```

Comment:
I'm not sure this is working as expected because I don't think you can match let-bound strings. 

```
module Const = struct
  let foo = "foo"
end

let f = function
  | foo -> print_endline "match foo"
  | _   -> print_endline "match something else"

let _ = f "bar"
```

Running this gives "match foo" for me.